typescript/utils/class-color-generator.ts

```diff
@@ -0,0 +1,110 @@
+export const hexColorSequence = [
+  "#F87171",
+  "#FACC15",
+  "#34D399",
+  "#38BDF8",
+  "#A78BFA",
+  "#DB2777",
+  "#FB923C",
+  "#65A30D",
+  "#0D9488",
+  "#2563EB",
+  "#9333EA",
+  "#FB7185",
+  "#FBBF24",
+  "#4ADE80",
+  "#22D3EE",
+  "#818CF8",
+  "#E879F9",
+  "#DC2626",
+  "#CA8A04",
+  "#059669",
+  "#0284C7",
+  "#7C3AED",
+  "#F472B6",
+  "#EA580C",
+  "#A3E635",
+  "#2DD4BF",
+  "#60A5FA",
+  "#C084FC",
+  "#E11D48",
+  "#D97706",
+  "#16A34A",
+  "#0891B2",
+  "#4F46E5",
+  "#C026D3",
+];
+
+// It's charkra's gray.200
+export const noneClassColor = "#E2E8F0";
+
+const createColorHashMap = (attibutedColors: string[]) => {
+  const hashMap: { [key: string]: number } = {};
+  attibutedColors.forEach((attributedColor) => {
+    if (attributedColor in hashMap) {
+      hashMap[attributedColor] += 1;
+    } else {
+      hashMap[attributedColor] = 1;
+    }
+  });
```

Comment:
According to my other comments, you are not sure here that `attibutedColors` will contain all the possible colors from `hexColorSequence`, and it will cause problems later.
You should first initialize the hashMap to zero for all `hexColorSequence` values.

Reply:
```suggestion
const countUniqueColors = (colors: string[]) => {
  return colors.reduce<Record<string, number>>(
    (counts, color) => ({
      ...counts,
      [color]: (counts[color] ?? 0) + 1,
    }),
    {}
  );
};
```